### DISCUSSION TITLE
Can't open HMI file in Nextion editor: "Wrong resource code"

### DISCUSSION THREAD

**Discussion by JaroslawPrzybyl** [HASwitchPlate/HASPone#177] [Q&A] [answered]
Because I have 3.5" model of display, I have to modify a HMI file to my own. Unfortunately I can't open it in Nextion editor, still there is an error: "Wrong resource file or resource file has been damaged". I'm trying with currently latest version of editor 1.63.3 and with LTS version (according to description based on 0.53 and dedicated for old codes) - still the same result. Please advise...

![image](https://user-images.githubusercontent.com/33628620/149148334-25d85ac1-ce3b-4abd-8ac7-cf6d6e0a6cb5.png)

**Comment by aderusha** [comment, marked as answer]
Sounds like maybe your HMI file is damaged, I can confirm that the copy currently in the repo does in fact open in 1.63.3 without warning so maybe try [re-downloading](https://github.com/HASwitchPlate/HASPone/blob/main/Nextion_HMI/HASwitchPlate.HMI) and trying again?

**Comment by JaroslawPrzybyl** [comment]
It's ashamed to admit, but stupid me downloaded this file not by obvious "download" but by right click and 'save as' :)
With right way everything works fine, thanks!

**Comment by vladtsvetkov880-hub** [comment]
Late to this, but I can explain what's actually going on, because I spent a while
inside this format.

The error message is misleading. The Editor reports a resource problem, but usually
the resources are fine. It fails earlier, when it opens the container. An `.HMI` has a
directory at the very start with a checksum right after the last entry, and the
container-open call returns NULL before the Editor has looked at a single image or
page. What you see is just the string it prints when opening fails, whatever the
reason was.

The practical consequence: if the file was touched by anything other than the Editor —
an interrupted download, a script, a git checkout — it's the directory that broke, not
the resources. Re-exporting images won't help, because images were never the problem.

Two things actually worth checking:

- **If the file went through git**, make sure it wasn't mangled on the way. `.HMI` is
  binary, and if it isn't marked as such in `.gitattributes`, line-ending conversion
  can quietly corrupt it on Windows.
- **Editor version.** Files from 0.53 and earlier genuinely need the LTS Editor. That
  one is a real incompatibility, not a corruption.

If you want to look inside a file and see whether the structure survived, I put a
read-only inspector here:

https://github.com/vladtsvetkov880-hub/nextion-hmi-format
